### PR TITLE
feat(ps): query k8s to get the ps:list information instead of db

### LINKED ISSUE #160: deis ps shows inaccurate process status
I have an application pod that is crashing:

```
[kent@mbp ~]$ k get pods --namespace=burley-autoharp
NAME                           READY     STATUS             RESTARTS   AGE
burley-autoharp-v2-web-6kilr   1/1       Running            0          2d
burley-autoharp-v2-web-fvcim   0/1       CrashLoopBackOff   7          15m
```

(Don't worry about _why_.  I made it crash.)

But `deis ps` doesn't accurately reflect this:

```
[kent@mbp ~]$ deis ps -a burley-autoharp
=== burley-autoharp Processes
--- web:
web.1 up (v2)
web.2 up (v2)
```

Apart from it being inaccurate that both are up, it's also impossible for me to know _which_ of these is wrong.  Which of those two pods is "web.1" and which is "web.2"?  There is no easy way to tell.  So I'd further propose that labeling processes in that manner (a carryover from the old fleet scheduler on v1.x) is not doing anyone any favors.

### PR DESCRIPTION
**Pull request by helgi**
Fetches all the information directly from k8s and uses the pod name in the output. Can no longer do a limit fetch as the k8s API does not support it

Fixes #160

### CODE REVIEW

**Comment by helgi** [comment]
```
deis create --no-remote
Creating Application... done, created jangly-playroom
remote available at ssh://git@deis.172.17.8.100.xip.io:2222/jangly-playroom.git
```

```
deis pull deis/example-go -a jangly-playroom
Creating build... done
```

Now `ps:list`
```
deis ps:list -a jangly-playroom
=== jangly-playroom Processes--- cmd:
jangly-playroom-v2-cmd-9yn9q up (v2)
```

Lets scale

```
deis ps:scale cmd=3 -a jangly-playroom
Scaling processes... but first, coffee!
done in 7s
=== jangly-playroom Processes--- cmd:
jangly-playroom-v2-cmd-9yn9q up (v2)
jangly-playroom-v2-cmd-ix5sx up (v2)
jangly-playroom-v2-cmd-mw33o up (v2)
```

Ask k8s what it things is the story

```
k get po --namespace jangly-playroom
NAME                           READY     STATUS    RESTARTS   AGE
jangly-playroom-v2-cmd-9yn9q   1/1       Running   0          1m
jangly-playroom-v2-cmd-ix5sx   1/1       Running   0          25s
jangly-playroom-v2-cmd-mw33o   1/1       Running   0          25s
```

Scale back

```
deis ps:scale cmd=2 -a jangly-playroom
Scaling processes... but first, coffee!
done in 5s
=== jangly-playroom Processes--- cmd:
jangly-playroom-v2-cmd-ix5sx up (v2)
jangly-playroom-v2-cmd-mw33o up (v2)
```

Talk to k8s again

```
k get po --namespace jangly-playroom
NAME                           READY     STATUS    RESTARTS   AGE
jangly-playroom-v2-cmd-ix5sx   1/1       Running   0          42s
jangly-playroom-v2-cmd-mw33o   1/1       Running   0          42s
```

Verify local

```
deis ps:list -a jangly-playroom
=== jangly-playroom Processes--- cmd:
jangly-playroom-v2-cmd-ix5sx up (v2)
jangly-playroom-v2-cmd-mw33o up (v2)
```

**Comment by slack** [comment]
Do we store the desired "scale" for a command type in the database?

If so, we should also show "desired" and "current".

**Comment by helgi** [comment]
We do not store those, no but I can pull that from the RC. That's a new functionality tho.

**Comment by aledbf** [comment]
> Can no longer do a limit fetch as the k8s API does not support it

Are you going to remove `LimitedRequest` then?

File: client/controller/models/ps/ps.go [hunk 1]
```diff
@@ -13,9 +13,8 @@ import (
 func List(c *client.Client, appID string, results int) ([]api.Process, int, error) {
 	u := fmt.Sprintf("/v2/apps/%s/containers/", appID)
 	body, count, err := c.LimitedRequest(u, results)
-
 	if err != nil {
-		return []api.Process{}, -1, err
+		return []api.Process{}, err
```
Comment (aledbf) [inline client/controller/models/ps/ps.go]: @helgi missing the `count` in the output

Comment (helgi) [inline client/controller/models/ps/ps.go]: Yeah this isn't ready. Not in a reviewable state

**Comment by helgi** [comment]
I brought `LimitedRequest` back since tests are being a pain. Working through that and it may change that all again

**Comment by helgi** [comment]
This is ready for review now. I made a new endpoint under `apps` instead of trying to replace the old one after chatting to @bacongobbler about it.

Keeping the pagination faking to keep the API at least unified for now.

**Comment by bacongobbler** [comment]
Once this is merged, we'll also need to create a new issue to tackle removing the Container object

File: rootfs/scheduler/mock.py [hunk 2]
```diff
@@ -57,6 +57,18 @@ def stop(self, name):
         job['state'] = JobState.down
 
     # Related to k8s services
+    def resolve_state(self, pod):
+        # See "Pod Phase" at http://kubernetes.io/v1.1/docs/user-guide/pod-states.html
+        states = {
+            "Pending": JobState.initialized,
```
Comment (slack) [inline rootfs/scheduler/mock.py]: Terminating should be on the list afaik...

Comment (helgi) [inline rootfs/scheduler/mock.py]: Can't see it in their docs http://kubernetes.io/v1.1/docs/user-guide/pod-states.html

Comment (helgi) [inline rootfs/scheduler/mock.py]: just tried it, terminating pods just don't show up in the API anymore. Did a quick look only tho

**Comment by slack** [comment]
Straaaange:
```
k:trout example-go [master]$ k --v=10 --namespace=inward-rucksack get po inward-rucksack-v2-web-aq8vx
I0211 11:25:55.711274   85272 debugging.go:102] curl -k -v -XGET  -H "User-Agent: kubectl/v1.1.3 (darwin/amd64) kubernetes/6a81b50" https://trout.slack.io/api
I0211 11:25:55.870082   85272 debugging.go:121] GET https://trout.slack.io/api 200 OK in 158 milliseconds
I0211 11:25:55.870148   85272 debugging.go:127] Response Headers:
I0211 11:25:55.870156   85272 debugging.go:130]     Date: Thu, 11 Feb 2016 19:25:56 GMT
I0211 11:25:55.870161   85272 debugging.go:130]     Content-Length: 57
I0211 11:25:55.870166   85272 debugging.go:130]     Content-Type: application/json
I0211 11:25:55.870233   85272 request.go:746] Response Body: {
  "kind": "APIVersions",
  "versions": [
    "v1"
  ]
}
I0211 11:25:55.871572   85272 debugging.go:102] curl -k -v -XGET  -H "User-Agent: kubectl/v1.1.3 (darwin/amd64) kubernetes/6a81b50" https://trout.slack.io/api/v1/namespaces/inward-rucksack/pods/inward-rucksack-v2-web-aq8vx
I0211 11:25:55.900439   85272 debugging.go:121] GET https://trout.slack.io/api/v1/namespaces/inward-rucksack/pods/inward-rucksack-v2-web-aq8vx 200 OK in 28 milliseconds
I0211 11:25:55.900468   85272 debugging.go:127] Response Headers:
I0211 11:25:55.900476   85272 debugging.go:130]     Content-Type: application/json
I0211 11:25:55.900483   85272 debugging.go:130]     Date: Thu, 11 Feb 2016 19:25:56 GMT
I0211 11:25:55.900526   85272 request.go:746] Response Body: {"kind":"Pod","apiVersion":"v1","metadata":{"name":"inward-rucksack-v2-web-aq8vx","generateName":"inward-rucksack-v2-web-","namespace":"inward-rucksack","selfLink":"/api/v1/namespaces/inward-rucksack/pods/inward-rucksack-v2-web-aq8vx","uid":"afbd5658-d0e0-11e5-b8ae-028241ade2f7","resourceVersion":"981281","creationTimestamp":"2016-02-11T16:58:34Z","deletionTimestamp":"2016-02-11T17:02:50Z","deletionGracePeriodSeconds":30,"labels":{"app":"inward-rucksack","heritage":"deis","type":"web","version":"v2"},"annotations":{"kubernetes.io/created-by":"{\"kind\":\"SerializedReference\",\"apiVersion\":\"v1\",\"reference\":{\"kind\":\"ReplicationController\",\"namespace\":\"inward-rucksack\",\"name\":\"inward-rucksack-v2-web\",\"uid\":\"d912b4b6-d0df-11e5-b8ae-028241ade2f7\",\"apiVersion\":\"v1\",\"resourceVersion\":\"980962\"}}\n"}},"spec":{"volumes":[{"name":"minio-user","secret":{"secretName":"minio-user"}},{"name":"default-token-y69rk","secret":{"secretName":"default-token-y69rk"}}],"containers":[{"name":"inward-rucksack-web","image":"quay.io/deisci/slugrunner:v2-beta","args":["start","web"],"env":[{"name":"PORT","value":"5000"},{"name":"SLUG_URL","value":"http://10.3.0.70:9000/git/home/inward-rucksack:git-71027744/push/slug.tgz"},{"name":"DEBUG","value":"1"},{"name":"DEIS_APP","value":"inward-rucksack"},{"name":"DEIS_RELEASE","value":"v2"},{"name":"DOCKERIMAGE","value":"1"}],"resources":{},"volumeMounts":[{"name":"minio-user","readOnly":true,"mountPath":"/var/run/secrets/object/store"},{"name":"default-token-y69rk","readOnly":true,"mountPath":"/var/run/secrets/kubernetes.io/serviceaccount"}],"livenessProbe":{"httpGet":{"path":"/","port":5000,"scheme":"HTTP"},"initialDelaySeconds":10,"timeoutSeconds":1},"readinessProbe":{"httpGet":{"path":"/","port":5000,"scheme":"HTTP"},"initialDelaySeconds":10,"timeoutSeconds":1},"terminationMessagePath":"/dev/termination-log","imagePullPolicy":"Always"}],"restartPolicy":"Always","terminationGracePeriodSeconds":30,"dnsPolicy":"ClusterFirst","serviceAccountName":"default","serviceAccount":"default","nodeName":"ip-10-0-0-212.us-west-2.compute.internal"},"status":{"phase":"Running","conditions":[{"type":"Ready","status":"True","lastProbeTime":null,"lastTransitionTime":null}],"hostIP":"10.0.0.212","podIP":"10.2.90.3","startTime":"2016-02-11T16:58:34Z","containerStatuses":[{"name":"inward-rucksack-web","state":{"running":{"startedAt":"2016-02-11T16:58:36Z"}},"lastState":{},"ready":true,"restartCount":0,"image":"quay.io/deisci/slugrunner:v2-beta","imageID":"docker://bfc3e9335c6bd79659da5db384cc887a277bbe3d085e6962d5073d2dbbcb7428","containerID":"docker://e41d5189c06d4f994a9a6ab6222a6b33824de6c87ff81999d0f794d3f7e12101"}]}}
NAME                           READY     STATUS        RESTARTS   AGE
inward-rucksack-v2-web-aq8vx   1/1       Terminating   0          2h
```

**Comment by slack** [comment]
Must be coming from `deletionTimestamp`...

File: rootfs/scheduler/mock.py (hunk 2 above)
Comment (bacongobbler) [inline rootfs/scheduler/mock.py]: the pod phases come from here: https://github.com/kubernetes/kubernetes/blob/b7dc1175ba64ce36a52fa44910ea7dcb283ffed9/pkg/api/types.go#L901-L919

I believe these are distinctly different compared to namespace phases: https://github.com/kubernetes/kubernetes/blob/4ca66d2aefa20c27b670b2fa890052daadc05294/docs/design/namespaces.md#phases

**Comment by mboersma** [comment]
This change to the REST API should also be updated in docs/src/reference-guide/controller-api/v2.0--although I see that isn't yet being included in the documentation set. I'll file a separate issue instead to bring the v2.0 API reference up to date.

**Comment by helgi** [comment]
Yeah we have done nothing to the API docs. I presumed they were auto generated, and if not then I'm going to install the official DRF auto generated docs in favour of anything handwritten.

**Comment by mboersma** [comment]
Works for me:
```console
$ deis ps -a kindly-yodeling
=== kindly-yodeling Processes
--- web:
web.1 up (v2)
$ deis scale -a kindly-yodeling web=6
Scaling processes... but first, Slurm!
done in 0s
=== kindly-yodeling Processes
--- web:
web.1 up (v2)
web.2 up (v2)
web.3 up (v2)
web.4 up (v2)
web.5 up (v2)
web.6 up (v2)
$ deis ps -a kindly-yodeling
=== kindly-yodeling Processes
--- web:
web.1 up (v2)
web.2 up (v2)
web.3 up (v2)
web.4 up (v2)
web.5 up (v2)
web.6 up (v2)
```

**Pull request merged**